lib/active_merchant/billing/gateways/card_stream.rb

```diff
@@ -283,6 +284,20 @@ def add_threeds_required(post, options)
         add_pair(post, :threeDSRequired, options[:threeds_required] || @threeds_required ? 'Y' : 'N')
       end
 
+      def add_global_threeds2_mpi_data(post, options)
+        return unless three_d_secure = options[:three_d_secure]
+
+        add_pair(post, :threeDSEnrolled, three_d_secure[:enrolled] == 'true' ? 'Y' : 'N')
+        if three_d_secure[:enrolled] == 'true'
+          add_pair(post, :threeDSAuthenticated, three_d_secure[:authentication_response_status])
+          if three_d_secure[:authentication_response_status] == 'Y'
+            post[:threeDSECI]  = three_d_secure[:eci]
+            post[:threeDSCAVV] = three_d_secure[:cavv]
+            post[:threeDSXID] = three_d_secure[:ds_transaction_id]
```

Comment:
🛑 Warning:

Here you need `three_d_secure[:xid]` param not ds_transaction_id, which belongs to 3DS v2.